mesonbuild/modules/gnome.py

```diff
@@ -848,20 +848,27 @@ def _get_build_args(self, kwargs, state, depends):
             if not isinstance(incd.held_object, (str, build.IncludeDirs)):
                 raise MesonException(
                     'Gir include dirs should be include_directories().')
+
         cflags.update(get_include_args(inc_dirs))
-        cflags.update(state.environment.coredata.get_external_args('c'))
         ldflags = OrderedSet()
         ldflags.update(internal_ldflags)
-        ldflags.update(state.environment.coredata.get_external_link_args('c'))
         ldflags.update(external_ldflags)
+
+        if state.environment.is_cross_build():
+            compiler = state.environment.coredata.cross_compilers.get('c')
```

Comment:
Should add args from the cross file here I think. ninjabackend.py takes them using `environment.cross_info.config['properties'][langname + '_link_args']` and `environment.cross_info.config['properties'][langname + '_args']`.

But then we are really just re-implementing backend's job here, so I think it's fine to leave it like that for now, and see if we can build the scanner with ninja instead.